framework/src/multiapps/MultiApp.C

```diff
@@ -502,6 +502,7 @@ MultiApp::createApp(unsigned int i, Real start_time)
   // Update the MultiApp level for the app that was just created
   app->setMultiAppLevel(_app.multiAppLevel() + 1);
   app->setupOptions();
+  preRunInputFile();
```

Comment:
Can a MooseApp know it's wrapped by a MultiApp? If so, we could do some special setup in a wrapped MooseApp based on the master app.